primitives/src/side_effect/mod.rs

```diff
@@ -28,10 +28,84 @@ pub type SideEffectId<T> = <T as frame_system::Config>::Hash;
 
 #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
 pub struct FullSideEffect<AccountId, BlockNumber, BalanceOf> {
-    pub input: SideEffect<AccountId, BlockNumber, BalanceOf>,
+    pub input: SideEffect<AccountId, BalanceOf>,
     pub confirmed: Option<ConfirmedSideEffect<AccountId, BlockNumber, BalanceOf>>,
     pub security_lvl: SecurityLvl,
     pub submission_target_height: Bytes,
+    pub best_bid: Option<SFXBid<AccountId, BalanceOf>>,
+}
+
+/// All Executors from the active set can bid for SFX executions in order to claim the rewards (max_fee) set by users,
+///     ultimately competing against one another on the open market rules.
+/// In case bid goes on Optimistic SFX, Executor will also have their bonded stake reserve to insure
+///     other Optimistic Executors co-executing given Xtx with their bonded collateral (reserved_bond)
+/// Their balance
+#[derive(Clone, Eq, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+pub struct SFXBid<AccountId, BalanceOf> {
+    /// Bid amount - always below SFX::max_fee requested by a user
+    pub bid: BalanceOf,
+    /// Insurance in case of optimistic FSX
+    pub insurance: BalanceOf,
+    /// Optional reserved bond in case of optimistic FSX
+    pub reserved_bond: Option<BalanceOf>,
+    /// Bidding Executor belonging to the active set
+    pub executor: AccountId,
+    /// Executor - subject of SFX
+    pub requester: AccountId,
+}
+
+impl<AccountId, BalanceOf> SFXBid<AccountId, BalanceOf> {
+    pub fn new_none_optimistic(
+        bid: BalanceOf,
+        insurance: BalanceOf,
+        executor: AccountId,
+        requester: AccountId,
+    ) -> Self {
+        SFXBid {
+            bid,
+            insurance,
+            reserved_bond: None,
+            executor,
+            requester,
+        }
+    }
+
+    pub fn expect_reserved_bond(&self) -> &BalanceOf {
+        self.reserved_bond
+            .as_ref()
+            .expect("Accessed reserved_bond and expected it to be a part of SFXBid")
+    }
+
+    pub fn get_insurance(&self) -> &BalanceOf {
+        &self.insurance
+    }
+}
+
+impl<AccountId, BlockNumber, BalanceOf> FullSideEffect<AccountId, BlockNumber, BalanceOf>
+where
+    AccountId: Encode + Clone,
+    BlockNumber: Encode + Clone,
+    BalanceOf: Encode + Zero + Clone,
+{
+    pub fn is_successfully_confirmed(&self) -> bool {
+        self.confirmed.is_some()
+            && self
+                .confirmed
+                .as_ref()
+                .expect("ensured exists in the same check")
+                .err
```

Comment:
Cant you just do `if let Some(confirmed) = confirmed { confirmed.err.is_none() } else { false }`

This would make it safe